ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/selectors/BlockSizeTransactionSelector.java

```diff
@@ -29,37 +29,39 @@
  * evaluating transactions based on block size. It checks if a transaction is too large for the
  * block and determines the selection result accordingly.
  */
-public class BlockSizeTransactionSelector extends AbstractTransactionSelector {
+public class BlockSizeTransactionSelector extends AbstractStatefulTransactionSelector<Long> {
```

Comment:
AbstractStatefulTransactionSelector. I feel that this name should be in conflict with Statefull and stateless worldstate. Just thniking I don't have a strong opinion about that

Reply:
I agree that _state_ can refer to different things here, so one should read it along with _transaction selector_ to fully qualify, not sure what could be another naming for this